consensus/consensus-types/src/safety_data.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::vote::Vote;
+use serde::{Deserialize, Serialize};
+use std::fmt;
+
+/// Data structure for safety rules to ensure consensus safety.
+#[derive(Debug, Deserialize, Eq, PartialEq, Serialize, Clone, Default)]
+pub struct SafetyData {
+    pub epoch: u64,
+    pub last_voted_round: u64,
+    pub preferred_round: u64,
+    pub last_vote: Option<Vote>,
```

Comment:
What does the serialised last_vote look like?

Reply:
"last_vote":{"vote_data":{"proposed":{"epoch":3,"round":10,"id":"c13f6b763a8f71c85990e5c8b0f359e35887b055765c4617e7f0c60c59a7601e","executed_state_id":"33bee8aadf5dacc829f5c419de23d44c3f7e21e0ee029d8e9b77bc358a4a7d8b","version":30,"timestamp_usecs":1597861702335358,"next_epoch_state":null},"parent":{"epoch":3,"round":9,"id":"e26f4c5bbd4840acee8c4945f4c698e278b617b67f4ee576d8ffda23f8e30a34","executed_state_id":"0c8e1a63ea54f267a6f45d9cd7e696f2dd043482989f2809807ef3925cab7726","version":29,"timestamp_usecs":1597861699689374,"next_epoch_state":null}},"author":"098e93340fe3ea57d0adad9d4c45b1ff","ledger_info":{"commit_info":{"epoch":3,"round":8,"id":"d4a39baa893d56b637789d95393f198f8377ac873d691f883f6f524bd6d20b79","executed_state_id":"793e748ceb7a8de7fdac42e30b3ff4c9416d8d35ab390f53ebfa43605d31766a","version":28,"timestamp_usecs":1597861699549195,"next_epoch_state":null},"consensus_data_hash":"4f3958b8fdccbcaa742d5e62abcccdc8c5321bd56bc6743d5d1f72d92e367eea"},"signature":"19d078fba8c4de0cd27daa739a7c666fb333249f59b9d8b12aca11f18943ab9f86db7b1e6b8007ac162c54d0cf7eef8171f545a4dd8dbfbcd580192e81e8820a","timeout_signature":null}

Reply:
Awesome, so not an array of integers :-D
What does an empty vote look like? I need this for the vault-init Terraform. Is it just `"last_vote": null` ?

Reply:
yup, it's null